dataprofiler/profilers/profile_builder.py

```diff
@@ -2287,7 +2287,11 @@ def _update_null_replication_metrics(self, clean_samples: Dict) -> None:
             # Partition data based on whether target column value is null or not
             # Calculate sum, mean of each partition without including current column
             # in calculation
-            sum_null = data.iloc[null_indices, data.columns != col_id].sum().to_numpy()
+            sum_null = (
+                data.loc[data.index.intersection(null_indices), data.columns != col_id]
+                .sum()
+                .to_numpy()
+            )
```

Comment:
This code does not error anymore if the entire DataFrame is null.

Reply:
this is much cleaner!

Reply:
beautiful!